frontend/src/modules/teamcomments/actions.js

```diff
@@ -59,9 +73,24 @@ export function get(entity: number, locale: string): Function {
     }
 }
 
+export function savePinnedStatus(pinned: boolean, commentId: number): Function {
+    return async dispatch => {
+        if (pinned) {
+            await api.comment.pinComment(commentId);
+        }
+        else {
+            await api.comment.unpinComment(commentId);
+        }
+
+        dispatch(updatePinned(pinned, commentId))
```

Comment:
I'd also use `toggle` instead of `update` here and in `UpdatePinnedAction` and `UPDATE_PINNED`.

I'd also call `updatePinnedComment` `togglePinnedStatus`.